core/src/main/scala/fs2/Chunk.scala

```diff
@@ -200,20 +288,105 @@ object Chunk {
       (0 until size).foldLeft(z)((z,i) => f(z, at(i)))
     def foldRight[B](z: B)(f: (Boolean,B) => B): B =
       ((size-1) to 0 by -1).foldLeft(z)((tl,hd) => f(at(hd), tl))
+
+    // here be dragons; modify with EXTREME care
+    override def map[B](f: Boolean => B): Chunk[B] = {
```

Comment:
**Update**:  So it appears that the semantic for *compiling* specialization with multiple specialized parameters (e.g. `Function1`) is that if *one* type parameter cannot be specialized, then *neither* are specialized.  In other words, there is no special type for `Long => Byte`, as can be verified somewhat easily:

```scala
{ x: Long => 0.toByte }.getClass.getSuperclass      // => AbstractFunction1
```

Now, the respecialization *is* working in cases where both parameters are specialized.  For example, `Long => Long`:

```
      38: invokevirtual #128                // Method values:()[J
      41: iload_2
      42: laload
      43: invokeinterface #318,  3          // InterfaceMethod scala/Function1.apply$mcJJ$sp:(J)J
      48: lastore
```

Basically, I think specialization is not treating "erased" as a permutation for specialization, which means that there's a lot of boxing going on that doesn't need to be.